// typescript-coverage-report/dist/lib/reporters/text.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
exports.generate = void 0;
var terminal_table_1 = __importDefault(require("terminal-table"));
require("colors");
var coverageTable = new terminal_table_1["default"]({
    leftPadding: 1,
    rightPadding: 1,
    borderStyle: 2
});
var calculatePercantage = function (correct, total) {
    if (total === 0) {
        return 100;
    }
    return (correct * 100) / total;
};
var calculatePercantageWithString = function (correct, total) {
    return calculatePercantage(correct, total).toFixed(2) + "%";
};
var generate = function (_a, threshold) {
    var fileCounts = _a.fileCounts, percentage = _a.percentage, total = _a.total, covered = _a.covered, uncovered = _a.uncovered;
    var row = 1;
    var headers = [
        "filenames" + (" (" + fileCounts.size + ")").gray,
        "percent" + (" (" + percentage.toFixed(2) + "%)").gray,
        "total" + (" (" + total + ")").gray,
        "covered" + (" (" + covered + ")").gray,
        "uncovered" + (" (" + uncovered + ")").gray
    ];
    coverageTable.push(headers, headers.map(function () { return "---".gray; }));
    coverageTable.attrRange({ column: [1, 5] }, {
        align: "right"
    });
    fileCounts.forEach(function (_a, filename) {
        var totalCount = _a.totalCount, correctCount = _a.correctCount;
        row++;
        coverageTable.push([
            filename,
            calculatePercantageWithString(correctCount, totalCount),
            totalCount,
            correctCount,
            totalCount - correctCount
        ]);
        coverageTable.attrRange({ row: [row] }, {
            color: Math.floor(calculatePercantage(correctCount, totalCount)) >=
                threshold
                ? "green"
                : "red"
        });
    });
    return "" + coverageTable;
};
exports.generate = generate;
